engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/SpecificRecordKafkaAvroSourceFactory.scala

```diff
@@ -27,8 +27,18 @@ class SpecificRecordKafkaAvroSourceFactory[T <: SpecificRecord: ClassTag](schema
                   (implicit processMetaData: MetaData, nodeId: NodeId): KafkaSource[T] = {
     val kafkaConfig = KafkaConfig.parseProcessObjectDependencies(processObjectDependencies)
     val preparedTopic = KafkaUtils.prepareKafkaTopic(topic, processObjectDependencies)
-    val schemaDeterminer = new SpecificRecordEmbeddedSchemaDeterminer(classTag[T].runtimeClass.asInstanceOf[Class[_ <: SpecificRecord]])
-    createSource(preparedTopic, kafkaConfig, schemaRegistryProvider.deserializationSchemaFactory, schemaRegistryProvider.recordFormatter, schemaDeterminer, returnGenericAvroType = false)
+    val valueSchemaDeterminer = new SpecificRecordEmbeddedSchemaDeterminer(classTag[T].runtimeClass.asInstanceOf[Class[_ <: SpecificRecord]])
+    createSource(
+      preparedTopic,
+      kafkaConfig,
+      schemaRegistryProvider.deserializationSchemaFactory,
+      schemaRegistryProvider.recordFormatter,
+      valueSchemaDeterminer,
+      keySchemaDeterminer = null,
```

Comment:
No no no, we are not using `null` in scala! Always use `Option` instead... But in this case IMO simpler would be to create case object `FixedNoneSchemaDeterminer`. WDYT?

Reply:
Fixed. Created `FixedNoneSchemaDeterminer`.